File: RansomNote.py
def canConstruct(ransomNote, magazine):
    wcRansom = {}
    wcMagazine = {}
    
    for letter in ransomNote:
        if letter not in wcRansom.keys(): wcRansom[letter] = 1
        else:
            wcRansom[letter] += 1
            
    for letter in magazine:
        if letter not in wcMagazine.keys(): wcMagazine[letter] = 1
        else:
            wcMagazine[letter] += 1
            
            
    for letter in wcRansom.keys():
        if letter not in wcMagazine.keys() or wcRansom[letter] < wcMagazing[letter]:
            return False
        
    return True

File: RansomNote.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    let map: Map<string, number> = new Map();

    for (let i=0; i < magazine.length; i++) {
        let char: string = magazine[i];
        if (map.has(char)) {
            map.set(char, map.get(char)! + 1);
        } else {
            map.set(char, 1);
        }
    }

    for (let i=0; i < ransomNote.length; i++) {
        let char: string = ransomNote[i];
        if (!map.has(char)) return false;
        else {
            map.set(char, map.get(char)! - 1);
            if (map.get(char)! < 0) return false;
        }
    }

    return true;
};